src/framework/core/js/ModelTransformationTransforms.js

```diff
@@ -653,12 +695,11 @@ var fluid = fluid || fluid_2_0_0;
         return parsedOffset;
     };
 
-    fluid.transforms.invertArrayIndexation = function (transformSpec, transformer) {
-        var togo = fluid.model.transform.copyInversePaths(transformSpec, transformer);
-        if (!isNaN(Number(togo.offset))) {
-            togo.offset = Number(togo.offset) * (-1);
+    fluid.transforms.invertArrayIndexation = function (transformSpec) {
```

Comment:
Now I see we need to be VERY VERY CLEAR because something quite subtle and probably highly dangerous has happened. The inversion path in the framework will now provide a copy of transformSpec to an fluid.standardTransformFunction inverter, by virtue of calling "copyInversePaths", but will provide the original transformSpec to other inverters. This is probably an unacceptable subtlety, and we should make sure to copy the transformSpec once during the other path too and clearly document this fact. All of this annoyance will go away with the IMMUTABLE REVOLUTION upcoming in the next revision of the core framework.

Once we have done this, we can eliminate the copy operation that inevitably occurs at the head of all the nonstandard transforms like valueMapper.

Also, the work in removing copyInversePaths is incomplete - we should be able to treat multiInput transforms as well, since the junk is still present in the inverse, e.g. of fluid.transforms.linearScale